pyteal/ast/router.py

```diff
@@ -724,6 +724,7 @@ def compile_program(
         version: int = DEFAULT_TEAL_VERSION,
         assemble_constants: bool = False,
         optimize: OptimizeOptions = None,
+        frame_pointers: Optional[bool] = None,
```

Comment:
Apologies if we already discussed and came to a different conclusion regarding how to handle the`frame_pointers` compilation behavior. However, should we consider a slightly different API? As this is a new optimization, the following may make sense. Instead of introducing a new variable for `router`, we can update `OptimizeOptions` and have it keep track of the frame_pointers usage. The update would look something like:

```python
class OptimizeOptions:
    def __init__(
        self, 
        *, 
        maximize: bool = True,
        scratch_slots: Optional[bool] = None,
        frame_pointers: Optional[bool] = None,
    ):
        self._skip_slots: Set[ScratchSlot] = set()

        if maximize:
            other_opts = []
            if scratch_slots is not None:
                other_opts.append("scratch_slots")
            if frame_pointers is not None:
                other_opts.append("frame_pointers")
            if other_opts:
                raise TealCompileError(f"When choosing to maximize OptimizeOptions, other parameters {other_opts} should not be provided.")
            self.scratch_slots = self.frame_pointers = True
            return

        # for backwards compatibility, default to self.scratch_slots == False:
        self.scratch_slots = bool(scratch_slots)

        self.frame_pointers = frame_pointers in [None, True]
```

If this seems like a reasonable way forward, I can open a PR.

Reply:
moving `frame_pointers` option to `OptimizeOptions` seems reasonable to me, while I have some thoughts on `maximize`.

On a syntax level, I think it would be more preferred to introduce class member to return an `OptimizeOptions` instance from a class method, like `OptimizeOptions.maximize()` or `OptimizeOptions.O3`, which feels better for me on triggering all optimizations, without introducing complication in class object initialization arguments.

Does the argument above sounds reasonable in some sense?

Reply:
Will prepare a PR.

Reply:
I just wanted to chime in and say I agree with the decision reached here. Please tag me on the PR when it's ready.

Reply:
FYI - after discussing with @michaeldiamant , I'll make this backwards compatible. We should consider changing the behavior to default to `maximize` optimization for AVM v $\geq$ 9